ui/ducks/send/send.js

```diff
@@ -1005,16 +1013,20 @@ const slice = createSlice({
         } else {
           recipient.error = null;
         }
-
-        if (
-          isSendingToken &&
-          isValidHexAddress(recipient.userInput) &&
-          (toChecksumAddress(recipient.userInput) in contractMap ||
-            checkExistingAddresses(recipient.userInput, tokens))
-        ) {
-          recipient.warning = KNOWN_RECIPIENT_ADDRESS_WARNING;
-        } else {
-          recipient.warning = null;
+        recipient.warning = null;
+        if (isSendingToken && isValidHexAddress(recipient.userInput)) {
+          // When useTokenDetection flag is true the tokenList contains tokens with non-checksum address from the dynamic token service api,
+          // When useTokenDetection flag is false the tokenList contains tokens with checksum addresses from contract-metadata.
+          // So the flag indicates whether the address of tokens currently on the tokenList is checksum or not.
+          const userInput = useTokenDetection
+            ? recipient.userInput
+            : toChecksumAddress(recipient.userInput);
+          if (
+            tokenAddressList.includes(userInput) ||
```

Comment:
Ok I think I know why this has been bugging me. I think rather than doing the checksum if useTokenDetection is true, we should use the isEqualCaseSensitive check here. The checkExistingAddresses already does a case insensitive check.

Reply:
[Q] doestokenAddressList contain checksum addresses? The code this is replacing used the checksumaddress to look up in contractMap which was keyed by checksum address.

Reply:
tokenAddressList will have checksumaddress when the user opt-out of the new token service in that scenario the tokenList will be holding the same data from contract-metadata. I am working on Alex's suggestion that way I can stick to the old code with minimal changes.